### ISSUE TITLE
Don't reload znc.pem on every client connection

### ISSUE THREAD

**Issue by DarthGandalf** [znc/znc#1215]
Need some way to poke ZNC to reload it instead.

**Comment by DarthGandalf** [comment]
One option is to use inotify

**Comment by muffeeee** [comment]
Just add an command like znc reloadcert or similar.

It will work for most purposes. Anyone who is using Letsencrypt (or any other CA) most likely already uses cron jobs to update the certificate and concatenate the cert files into znc.pem, so adding another cron job that reloads the cert after updating it shouldn't be an issue.

**Comment by DarthGandalf** [comment]
@muffeeee such command would require user to connect to ZNC via IRC protocol (and authenticate as some admin) from crontab.

**Comment by muffeeee** [comment]
No, I'm not talking about an IRC command - I'm talking about a UNIX command that can also be run manually from a terminal session.

`znc reloadcert`

Either that, or make `service znc reload` also reload cert, or make something like `service znc reloadcert`

**Comment by teward** [comment]
@muffeeee The second part only works when ZNC is set up as a service which is NOT the default configuration.  The first operates on the principle that there's a way to pass a command to the running ZNC from the command like, which I don't think is possible as it currently exists.

**Comment by muffeeee** [comment]
Either way, my znc doesn't reload the cert on every connection anyway for some reason, so I'm still stuck with a soon-to-expire cert unless I restart znc.

**Comment by DarthGandalf** [comment]
No permissions to read the file?
ZNC does NOT cache cert in memory yet. This bug is exactly about this.

2016-01-17 14:50 GMT+00:00 muffe <notifications@github.com>:

**Comment by Celti** [comment]
You could make ZNC reload all config files including znc.pem on SIGHUP, SIGUSR1, or SIGRTMIN.

**Comment by dequis** [comment]
Is this bug about a performance improvement? Is there a significant impact? When I first read that znc reloaded it on every connection it seemed like a great idea to me.

**Comment by Mikaela** [comment]
It's about LetsEncrypt, no one cared before it came.

**Comment by dequis** [comment]
Yeah but the current behavior makes it *easier* to use it with letsencrypt (no need to do anything to reload certs), while this ticket is about not reloading it every time and possibly adding an extra manual reload step (which is how it works everywhere else, but still)

So I assume it's for performance reasons. Just wondering how big of a deal that is. I don't know znc's scalability requirements, either.

**Comment by Mikaela** [comment]
From what I have understood the longer term aim is to

1. Start ZNC as root
2. ZNC reads certs from /etc/letsencrypt/whatever 
3.ZNC becomes the normal user and continues normally. 

Or at least that has been wished without having to manually copy LE certs around the system and lose automatic renewals.

**Comment by DarthGandalf** [comment]
> Start ZNC as root

Certainly no, see #1233 

> which is how it works everywhere else

That's the reason

**Comment by Nodens-** [comment]
If it's not done for performance benefit there is no reason to fix this. I consider it a feature.
And frankly the performance penalty of the file read is negligible. It's not a web server with thousands of connections requesting it constantly.
Otherwise triggering the reload of the certificate via handling SIGHUP is the best option.
Restarting znc to reload the certificate should not be considered.
